dotCMS/src/main/java/com/dotcms/security/secret/ServiceIntegrationAPIImpl.java

```diff
@@ -0,0 +1,165 @@
+package com.dotcms.security.secret;
+
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.LayoutAPI;
+import com.dotmarketing.business.UserAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.UtilMethods;
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+import io.vavr.Tuple2;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class ServiceIntegrationAPIImpl implements ServiceIntegrationAPI {
+    private static final String HOST_SECRET_KEY_SEPARATOR = ":";
+    private static final String DOT_GLOBAL_SERVICE = "dotCMSGlobalService";
+    private static final String INTEGRATIONS_PORTLET_ID = "integrationsPortlet";
+
+    private UserAPI userAPI;
+    private LayoutAPI layoutAPI;
+    private SecretsStore secretsStore;
+
+    private ObjectMapper objectMapper = new ObjectMapper()
+            //.enable(SerializationFeature.INDENT_OUTPUT)
+            .setVisibility(PropertyAccessor.FIELD, Visibility.ANY)
+    ;
+
+    private String toJsonString(final ServiceSecrets object) throws DotDataException {
+        try {
+            return objectMapper.writeValueAsString(object);
+        } catch (IOException e) {
+            throw new DotDataException(e);
+        }
+    }
+
+    private ServiceSecrets readJson(final String json) throws DotDataException {
+        try {
+            return objectMapper.readValue(json, ServiceSecrets.class);
+        } catch (IOException e) {
+            throw new DotDataException(e);
+        }
+    }
+
+    private ServiceSecrets readJson(final char [] chars) throws DotDataException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4623512)